src/main/core/support/units.rs

```diff
@@ -0,0 +1,703 @@
+use std::convert::TryInto;
+use std::fmt::{self, Debug, Display};
+use std::str::FromStr;
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+use serde::de::{Deserialize, Deserializer, Visitor};
+use serde::ser::{Serialize, Serializer};
+
+pub trait Prefixes: Clone + Copy + Default + PartialEq + FromStr + Display + Debug {
```

Comment:
Throughout - consider adding some doc comments to the `pub` elements.

Reply:
Added some more comments.